src/lib/api/user/client/userApiClient.client.ts

```diff
@@ -0,0 +1,40 @@
+'use client';
+
+import { request } from '@/lib/api/common/client';
+import { IUser } from '@/types/definition';
+
+import { Err, Ok, Result } from 'result-ts-type';
+
+const API_BASE_URL = process.env.NEXT_PUBLIC_API_BASE_URL || '';
+
+/**
+ * Interface representing the request body for the function to update user's data.
+ * @property name - The name of the user.
+ */
```

Comment:
Could you rewrite the TSDoc, please? This is not the official way.
#247 

Instead, please add a description to the line right above each property.